app/Jobs/Brevo/ProcessBrevoWebhook.php

```diff
@@ -0,0 +1,492 @@
+
+        // if(config('ninja.notification.slack'))
+        // $this->invitation->company->notification(new EmailBounceNotification($this->invitation->company->account))->ninja();
+    }
+
+    // {
+    //   "event" :  "spam",
+    //   "email" :  "example@example.com",
+    //   "id" :  1,
+    //   "date" :  "yyyy-mm-dd hh:i:s",
+    //   "message-id" :  "<xxx@msgid.domain>",
+    //   "tag" :  "<defined-tag>",//json of array
+    //   "tags": [
+    //        "company_key"
+    //    ],
+    //   "sending_ip" :  "xxx.xx.xxx.xx",
+    //   "sender_email": "user@example.com",
+    //  }
+    private function processSpamComplaint()
+    {
+        $this->invitation->email_status = 'spam';
+        $this->invitation->save();
+
+        $spam = new EmailSpam(
+            $this->request['tags'][0],
+            $this->request['sender_email'],
+            $this->request['message-id']
+        );
+
+        LightLogs::create($spam)->send();
+
+        $data = array_merge($this->request, ['history' => $this->fetchMessage()]);
+
+        $sl = $this->getSystemLog($this->request['message-id']);
+
+        if ($sl) {
+            $this->updateSystemLog($sl, $data);
+            return;
+        }
+
+        (new SystemLogger($data, SystemLog::CATEGORY_MAIL, SystemLog::EVENT_MAIL_SPAM_COMPLAINT, SystemLog::TYPE_WEBHOOK_RESPONSE, $this->invitation->contact->client, $this->invitation->company))->handle();
+
+        if (config('ninja.notification.slack')) {
+            $this->invitation->company->notification(new EmailSpamNotification($this->invitation->company->account))->ninja();
+        }
+    }
+
+    private function discoverInvitation(string $message_id)
+    {
+        $invitation = false;
+
+        if ($invitation = InvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'invoice';
+            return $invitation;
+        } elseif ($invitation = QuoteInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'quote';
+            return $invitation;
+        } elseif ($invitation = RecurringInvoiceInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'recurring_invoice';
+            return $invitation;
+        } elseif ($invitation = CreditInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'credit';
+            return $invitation;
+        } elseif ($invitation = PurchaseOrderInvitation::where('message_id', $message_id)->first()) {
+            $this->entity = 'purchase_order';
+            return $invitation;
+        } else {
+            return $invitation;
+        }
+    }
+
+    public function getRawMessage(string $message_id)
+    {
+
+        $brevo_secret = $this->company->settings->brevo_secret ?? config('services.brevo.key');
```

Comment:
left side will always resolve true for this.

Reply:
fixes